subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/JavaCompilerIntegrationSpec.groovy

```diff
@@ -58,6 +58,26 @@ abstract class JavaCompilerIntegrationSpec extends BasicJavaCompilerIntegrationS
         file("build/classes/main/compile/test/Person.class").exists()
     }
 
+    def compileWithCustomDiagnosticListener() {
```

Comment:
Could you also add relevant tests to `CompileOptionsTest` so we have unit test coverage e.g. in `testCompileOptions()` and new test case for the introduced option?